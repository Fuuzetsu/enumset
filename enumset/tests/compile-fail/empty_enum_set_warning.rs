#![deny(deprecated)]

use enumset::*;

#[derive(Debug, EnumSetType)]
enum Test {
    A, B, C
}

fn main() {
    let set: EnumSet<Test> = enum_set!();
    println!("{:?}", set);
}