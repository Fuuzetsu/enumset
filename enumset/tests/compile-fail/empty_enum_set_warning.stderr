error: use of deprecated function `enumset::__internal::empty_macro_deprecation`: The empty `enum_set!()` macro is deprecated, because it does not work with sets that use an array representation. Use `EnumSet::EMPTY` instead.
  --> tests/compile-fail/empty_enum_set_warning.rs:11:30
   |
11 |     let set: EnumSet<Test> = enum_set!();
   |                              ^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/compile-fail/empty_enum_set_warning.rs:1:9
   |
1  | #![deny(deprecated)]
   |         ^^^^^^^^^^
   = note: this error originates in the macro `enum_set` (in Nightly builds, run with -Z macro-backtrace for more info)
